nav2_collision_monitor/src/polygon.cpp

```diff
@@ -345,6 +379,80 @@ bool Polygon::getParameters(std::string & polygon_pub_topic, std::string & footp
   return true;
 }
 
+bool Polygon::getTransform(
```

Comment:
The similar function was used in data sources `Source::getTransform()`, but it is interpolating frames using time shifts between them. [Moved](https://github.com/ros-planning/navigation2/pull/3245/commits/b94bcc48e3e24d697cf04903c4d5c1e46ad29236) both functions to `nav2_util/robot_utils.hpp`, since it already has similar transformation routine.